### PR TITLE
Hammasi

### PR DESCRIPTION
**Pull request by AlisherDe**
Hammasi bajarildi

### CODE REVIEW

File: webapp/Controllers/AccountController.cs [hunk 1]
```diff
@@ -1,39 +1,105 @@
...
+        UserManager<AppUser> userManager,
+        SignInManager<AppUser> signinManager)
     {
+        _userManager = userManager;
+        _signinManager = signinManager;
         _logger = logger;
     }
     [HttpGet]
-    public IActionResult Register(string returnUrl) 
+    public IActionResult Signup(string returnUrl)
```
Comment (wahid-d) [inline webapp/Controllers/AccountController.cs:24]: Register

File: webapp/Controllers/AccountController.cs [hunk 2]
```diff
@@ -1,39 +1,105 @@
...
         _logger = logger;
     }
     [HttpGet]
-    public IActionResult Register(string returnUrl) 
+    public IActionResult Signup(string returnUrl) 
         => View(new RegisterViewModel() { ReturnUrl = returnUrl ?? string.Empty });
 
     [HttpPost]
-    public async Task<IActionResult> Register(RegisterViewModel model)
+    public async Task<IActionResult> Signup(RegisterViewModel model)
```
Comment (wahid-d) [inline webapp/Controllers/AccountController.cs:28]: Register

File: webapp/Controllers/AccountController.cs [hunk 3]
```diff
@@ -1,39 +1,105 @@
...
+        var user = new AppUser()
+        {
+            Fullname = model.Fullname,
+            Email = model.Email,
+            PhoneNumber = model.Phone,
+            UserName = model.Email.Substring(0, model.Email.IndexOf('@'))
+        };
+
+        await _userManager.CreateAsync(user, model.Password);
+        await _userManager.AddToRoleAsync(user, "student");
```
Comment (wahid-d) [inline webapp/Controllers/AccountController.cs:56]: Qanaqa student role?

**Review by wahid-d — CHANGES_REQUESTED** [review]

File: webapp/Controllers/AccountController.cs (hunk 3 above)
Comment (AlisherDe) [reply webapp/Controllers/AccountController.cs]: register controller
login controller

Comment (AlisherDe) [reply webapp/Controllers/AccountController.cs]: seed uchdi

Comment (AlisherDe) [reply webapp/Controllers/AccountController.cs]: uzr

File: webapp/Program.cs [hunk 4]
```diff
@@ -9,15 +12,30 @@
...
+.AddEntityFrameworkStores<AppDbContext>();
+
 // Add services to the container.
 builder.Services.AddControllersWithViews();
 
 builder.Services.ConfigureApplicationCookie(options =>
 {
     options.Cookie.Name = "invoice.app.identity";
+    options.LoginPath = "/account/login";
+    options.LogoutPath = "/account/register";
```
Comment (wahid-d) [inline webapp/Program.cs:33]: /account/logout

File: webapp/Controllers/AccountController.cs [hunk 5]
```diff
@@ -28,12 +64,25 @@ public IActionResult Login(string returnUrl)
...
+        if(!ModelState.IsValid)
+        {
+            return View(model);
+        }
 
-    [HttpPost]
-    public async Task<IActionResult> Logout()
-    {
-        return LocalRedirect("/");
+        var user = _userManager.Users.FirstOrDefault(u => u.Email == model.Email);
```
Comment (wahid-d) [inline webapp/Controllers/AccountController.cs:72]: FirstOrDefaultAsync()

**Review by wahid-d — CHANGES_REQUESTED** [review]

**Pull request merged**